doc/admin-guide/plugins/header_rewrite.en.rst

```diff
@@ -935,6 +943,16 @@ cache updates may have been performed. This hook context provides a means to
 modify aspects of the response sent to a client, while still caching the
 original versions of those attributes delivered by the origin server.
 
+TXN_START_HOOK
+~~~~~~~~~~~~~~
+
+Rulesets are evaluated when |TS| receives a request and accepts it. This hook context indicates that a HTTP transaction is initiated.
```

Comment:
At a minimum, I think we need to document that this would only work as a global plugin.